amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/TestConsumer.java

```diff
@@ -50,33 +43,87 @@ public class TestConsumer {
     private LeaseManagementConfig leaseManagementConfig;
     private LifecycleConfig lifecycleConfig;
     private ProcessorConfig processorConfig;
+    private Scheduler scheduler;
+    private ScheduledExecutorService producerExecutor;
+    private ScheduledFuture<?> producerFuture;
+    private DynamoDbAsyncClient dynamoClient;
     public int successfulPutRecords = 0;
     public BigInteger payloadCounter = new BigInteger("0");
 
-    public TestConsumer(KCLAppConfig consumerConfig) {
+    public TestConsumer(KCLAppConfig consumerConfig) throws Exception {
         this.consumerConfig = consumerConfig;
         this.region = consumerConfig.getRegion();
         this.streamName = consumerConfig.getStreamName();
-        this.kinesisClient = KinesisClientUtil.createKinesisAsyncClient(KinesisAsyncClient.builder().region(this.region));
+        this.kinesisClient = consumerConfig.buildAsyncKinesisClient(consumerConfig.getConsumerProtocol());
```

Comment:
have this protocal in the config itself which you are creating based on your test config